src/sagemaker_sklearn_extension/preprocessing/encoders.py

```diff
@@ -620,3 +621,248 @@ def inverse_transform(self, X):
                 X_tr[unknown_mask, idx] = None
 
         return X_tr
+
+
+class WOEAsserts(Enum):
+    ALPHA = "Regularization parameter `alpha` must be non-negative."
+    BINARY = "Weight-of-Evidence encoder is only supported for binary targets."
+    BINNING = "Binning strategy must be in {'uniform', 'quantile', 'kmeans'}."
+    NBINS = "Number of bins must be larger than 2."
+    MISSING = "Weight-of-Evidence encoder does not support missing values."
+    DIMS_TRANSFORM = "The number of columns of `X` doesn't match the encoder."
+    FEAT_OOR = "At least one of the feature index is out of the allowed range."
+
+
+class WOEEncoder(BaseEstimator, TransformerMixin):
+    """Weight of Evidence (WoE) encoder: encodes categorical features as a
+    numerical vector using weight of evidence encoding. This is only supported
+    with binary targets.
+    Both the features and the target are assumed to be free of missing values,
+    missing values should be handled separately before the encoding.
+    A binning function can be provided to handle numerical features which are
+    then binned first then encoded.
+    Note that the sign of the weight of evidence values depends on the
+    order in which the categories of the target column are detected. This does
+    not affect the performance of any supervised model applied thereafter.
+    See [1] for more details on WoE.
+
+    Parameters
+    ----------
+    feature_indices: list of integer indices (default=None)
+        Indices of the features to encode with WoE. If set to `None`, all
+        features will be encoded.
```

Comment:
This is unnecessary, the column selection functionality should be used by the [`ColumnTransformer`](https://scikit-learn.org/stable/modules/generated/sklearn.compose.ColumnTransformer.html)

Reply:
@jeanfad, I don't mind dropping this.